test_bot.py

```diff
@@ -0,0 +1,203 @@
+
+    return voice_channel
+
+
+def create_mock_voice_state(channel=None):
+    voice_state = mock.Mock()
+    voice_state.channel = channel
+    return voice_state
+
+
+def create_mock_author(name="default_author", voice_state=None):
+    author = mock.Mock()
+    author.author_str = name
+
+    def author_eq(self, other_author):
+        return self.author_str == other_author.author_str
+
+    author.__eq__ = author_eq
+
+    def author_repr(self):
+        return self.author_str
+
+    author.__repr__ = author_repr
+
+    author.voice = voice_state
+
+    return author
+
+
+def create_mock_message(
+    contents="",
+    author=create_mock_author(name="default_user"),
+):
+    message = mock.Mock()
+    message.content = contents
+    message.author = author
+    message.channel.send = mock.AsyncMock()
+
+    return message
+
+
+class MusicBotTest(unittest.IsolatedAsyncioTestCase):
+    """MusicBot test suite"""
+
+    async def asyncSetUp(self):
+        # pylint: disable=attribute-defined-outside-init
+
+        self.dispatcher_ = mock.Mock()
+        self.dispatcher_.user = create_mock_author(name="test_bot")
+        self.dispatcher_.loop = asyncio.get_running_loop()
+
+        self.guild_ = mock.Mock()
+
+        self.music_bot_ = bot.MusicBot(
+            self.guild_, self.dispatcher_.loop, self.dispatcher_.user
+        )
+        self.music_bot_.pafy_search = mock.Mock()
+        self.music_bot_.youtube_search = mock.MagicMock()
+        self.mock_audio_source_ = mock.Mock()
+        self.music_bot_.create_audio_source = mock.Mock(
+            return_value=self.mock_audio_source_
+        )
+
+    async def test_ignores_own_message(self):
+        message = create_mock_message(
+            contents="-Some bot message", author=self.dispatcher_.user
+        )
+
+        await self.music_bot_.handle_message(message)
+
+        # Bot didn't respond with anything.
+        message.channel.send.assert_not_awaited()
+
+    async def test_ignores_message_without_command_prefix(self):
+        ignore_message = create_mock_message(contents="Some non-command message")
+
+        await self.music_bot_.handle_message(ignore_message)
+
+        # Bot didn't respond with anything.
+        ignore_message.channel.send.assert_not_awaited()
+
+    async def test_hello_command_sends_message(self):
+        hello_message = create_mock_message(contents="-hello")
+
+        await self.music_bot_.handle_message(hello_message)
+
+        hello_message.channel.send.assert_awaited_with(":wave: Hello! default_user")
+
+    async def test_play_fails_when_user_not_in_voice_channel(self):
+        play_message = create_mock_message(
+            contents="-play song", author=create_mock_author()
+        )
+
+        await self.music_bot_.handle_message(play_message)
+
+        play_message.channel.send.assert_awaited_with(
+            ":studio_microphone: default_author, please join a voice channel to start the :robot:"
```

Comment:
This line can be modified to dynamically insert the user

Reply:
I think that goes against the spirit of testing. The test should be as transparent as possible.